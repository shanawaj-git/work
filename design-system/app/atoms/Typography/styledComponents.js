import { styled } from '@mui/material/styles';

import { Typography as TypographyMaterialUi } from '@mui/material';
import { colorSelector } from 'themes';

export const StyledTypography = styled(TypographyMaterialUi)(() => ({
  '&.MuiTypography-root': {
    color: colorSelector('fontPrimary'),
    fontFamily: 'Poppins',
  },
  '&.MuiTypography-h1': {
    fontStyle: 'normal',
    fontWeight: 'bold',
    fontSize: '34px',
    lineHeight: '41px',
    letterSpacing: '0.374px',
  },

  '&.MuiTypography-h2': {
    fontWeight: 'normal',
    fontSize: '28px',
    lineHeight: '34px',
  },

  '&.MuiTypography-h3': {
    fontWeight: 'normal',
    fontSize: '22px',
    lineHeight: '28px',
  },

  '&.MuiTypography-h4': {
    fontWeight: 'normal',
    fontSize: '20px',
    lineHeight: '24px',
  },

  '&.MuiTypography-h5': {
    fontWeight: '600',
    fontSize: '17px',
    lineHeight: '22px',
  },

  '&.MuiTypography-h6': {
    fontWeight: 'normal',
    fontSize: '16px',
    lineHeight: '21px',
  },

  '&.MuiTypography-caption': {
    fontWeight: 'normal',
    fontSize: '12px',
    lineHeight: '16px',
  },
}));
