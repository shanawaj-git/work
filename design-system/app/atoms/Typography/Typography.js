import React from 'react';
import PropTypes from 'prop-types';
import { StyledTypography } from './styledComponents';

export const TypographyType = Object.freeze({
  LARGE_TITLE: 'largeTitle',
  TITLE_1: 'title1',
  TITLE_2: 'title2',
  TITLE_3: 'title3',
  HEAD_LINE: 'headLine',
  CALL_OUT: 'callOut',
  SUB_HEAD: 'subHead',
  FOOT_NOTE: 'footNote',
  CAPTION: 'caption',
});

const mapTypographyTypeToVarient = {
  [TypographyType.LARGE_TITLE]: 'h1',
  [TypographyType.TITLE_1]: 'h2',
  [TypographyType.TITLE_2]: 'h3',
  [TypographyType.TITLE_3]: 'h4',
  [TypographyType.HEAD_LINE]: 'h5',
  [TypographyType.CALL_OUT]: 'h6',
  [TypographyType.SUB_HEAD]: 'body1',
  [TypographyType.FOOT_NOTE]: 'body2',
  [TypographyType.CAPTION]: 'caption',
};

const Typography = props => {
  const { children, typographyType, ...otherProps } = props;
  return (
    <StyledTypography
      {...otherProps}
      variant={mapTypographyTypeToVarient[typographyType]}
    >
      {children}
    </StyledTypography>
  );
};

Typography.TypographyType = TypographyType;

Typography.propTypes = {
  children: PropTypes.any.isRequired,
  typographyType: PropTypes.string,
};

export default Typography;
