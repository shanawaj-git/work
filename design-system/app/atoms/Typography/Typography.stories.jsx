import React from 'react';
import Typography, { TypographyType } from '.';
import StoryWrapper from 'molecules/utility/StoryWrapper';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Tokens/React/Typography',
  component: Typography,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    themeType: {
      control: { type: 'select', options: ['DEFAULT', 'MPC'] },
    },
    typographyType: {
      control: { type: 'select', options: Object.values(TypographyType) },
    },
    align: {
      control: {
        type: 'select',
        options: ['center', 'inherit', 'justify', 'left', 'right'],
      },
    },
  },
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = args => (
  <StoryWrapper themeVariant={args.themeType}>
    <Typography {...args}>This is a typography components</Typography>
  </StoryWrapper>
);

export const Title1 = Template.bind({});
Title1.args = {
  typographyType: TypographyType.TITLE_1,
  color: '',
  align: '',
};

export const Title2 = Template.bind({});
Title2.args = {
  typographyType: TypographyType.TITLE_2,
  color: '',
  align: '',
};

export const Title3 = Template.bind({});
Title3.args = {
  typographyType: TypographyType.TITLE_3,
  color: '',
  align: '',
};

export const Caption = Template.bind({});
Caption.args = {
  typographyType: TypographyType.CAPTION,
  color: '',
  align: '',
};

export const Callout = Template.bind({});
Callout.args = {
  typographyType: TypographyType.CALL_OUT,
  color: '',
  align: '',
};

export const Headline = Template.bind({});
Headline.args = {
  typographyType: TypographyType.HEAD_LINE,
  color: '',
  align: '',
};
