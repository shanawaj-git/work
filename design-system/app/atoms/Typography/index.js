import Typography, { TypographyType } from './Typography';

export { TypographyType };

export default Typography;
