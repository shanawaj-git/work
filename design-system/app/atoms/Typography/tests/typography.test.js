import React from 'react';
import renderer from 'react-test-renderer';

import Typography, { TypographyType } from '../index';

describe('Typography.js', () => {
  describe('index', () => {
    const testTypo = 'test typo';
    it('Should import callout typo', () => {
      const tree = renderer.create(
        <Typography typographyType={TypographyType.CALL_OUT}>
          {testTypo}
        </Typography>,
      );
      expect(tree).toBeTruthy();
      expect(tree).toMatchSnapshot();
    });

    it('Should import caption typo', () => {
      const tree = renderer.create(
        <Typography typographyType={TypographyType.CAPTION}>
          {testTypo}
        </Typography>,
      );
      expect(tree).toBeTruthy();
      expect(tree).toMatchSnapshot();
    });

    it('Should import foot note typo', () => {
      const tree = renderer.create(
        <Typography typographyType={TypographyType.FOOT_NOTE}>
          {testTypo}
        </Typography>,
      );
      expect(tree).toBeTruthy();
      expect(tree).toMatchSnapshot();
    });

    it('Should import head line typo', () => {
      const tree = renderer.create(
        <Typography typographyType={TypographyType.HEAD_LINE}>
          {testTypo}
        </Typography>,
      );
      expect(tree).toBeTruthy();
      expect(tree).toMatchSnapshot();
    });

    it('Should import large title typo', () => {
      const tree = renderer.create(
        <Typography typographyType={TypographyType.LARGE_TITLE}>
          {testTypo}
        </Typography>,
      );
      expect(tree).toBeTruthy();
      expect(tree).toMatchSnapshot();
    });

    it('Should import subtitile typo', () => {
      const tree = renderer.create(
        <Typography typographyType={TypographyType.SUB_HEAD}>
          {testTypo}
        </Typography>,
      );
      expect(tree).toBeTruthy();
      expect(tree).toMatchSnapshot();
    });

    it('Should import title 1 typo', () => {
      const tree = renderer.create(
        <Typography typographyType={TypographyType.TITLE_1}>
          {testTypo}
        </Typography>,
      );
      expect(tree).toBeTruthy();
      expect(tree).toMatchSnapshot();
    });

    it('Should import title 2 typo', () => {
      const tree = renderer.create(
        <Typography typographyType={TypographyType.TITLE_2}>
          {testTypo}
        </Typography>,
      );
      expect(tree).toBeTruthy();
      expect(tree).toMatchSnapshot();
    });

    it('Should import title 3 typo', () => {
      const tree = renderer.create(
        <Typography typographyType={TypographyType.TITLE_3}>
          {testTypo}
        </Typography>,
      );
      expect(tree).toBeTruthy();
      expect(tree).toMatchSnapshot();
    });
  });
});
