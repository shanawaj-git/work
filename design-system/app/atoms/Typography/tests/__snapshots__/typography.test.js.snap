// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography.js index Should import callout typo 1`] = `
<h6
  className="MuiTypography-root MuiTypography-h6 css-1t9yafy-MuiTypography-root"
>
  test typo
</h6>
`;

exports[`Typography.js index Should import caption typo 1`] = `
<span
  className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
>
  test typo
</span>
`;

exports[`Typography.js index Should import foot note typo 1`] = `
<p
  className="MuiTypography-root MuiTypography-body2 css-ry76j1-MuiTypography-root"
>
  test typo
</p>
`;

exports[`Typography.js index Should import head line typo 1`] = `
<h5
  className="MuiTypography-root MuiTypography-h5 css-11estw0-MuiTypography-root"
>
  test typo
</h5>
`;

exports[`Typography.js index Should import large title typo 1`] = `
<h1
  className="MuiTypography-root MuiTypography-h1 css-qu1isr-MuiTypography-root"
>
  test typo
</h1>
`;

exports[`Typography.js index Should import subtitile typo 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 css-1reifv3-MuiTypography-root"
>
  test typo
</p>
`;

exports[`Typography.js index Should import title 1 typo 1`] = `
<h2
  className="MuiTypography-root MuiTypography-h2 css-1jfg05l-MuiTypography-root"
>
  test typo
</h2>
`;

exports[`Typography.js index Should import title 2 typo 1`] = `
<h3
  className="MuiTypography-root MuiTypography-h3 css-1ewfuka-MuiTypography-root"
>
  test typo
</h3>
`;

exports[`Typography.js index Should import title 3 typo 1`] = `
<h4
  className="MuiTypography-root MuiTypography-h4 css-1b8g8n0-MuiTypography-root"
>
  test typo
</h4>
`;
